andela/ah-robotics-frontend
# #162949047 User should be able to follow and unfollow another

#### What does this PR do?
Enable user to be able to follow and or unfollow other users.
To enable a user a view to managing those they follow and also see those who follow them.
#### Description of Task to be completed?
#### How should this be manually tested?
1. Navigate to a folder where you would prefer to test the repository.

2. Create a directory
 ```$```  ```mkdir myproject```

3. Navigate to the directory "myproject"
 ```$```   ```cd myproject```
4. Clone the author's haven repository
  ```$```   ```git clone https://github.com/andela/ah-robotics-frontend.git```
5. Change the directory into the "ah-robotics-fronted"
  ```$```   ```cd ah-robotics-frontend```
6.  Checkout to the branch ft-follow-other-user-162949047
 ```$```   ```git checkout ft-follow-other-user-162949047```
7. Install dependancies from the package.json
 ```$```   ```npm install``` to install all the modules listed in the `package.json` file.

8. Start the application in development mode
  ```$```   ```npm start``` to run the app in the development mode 

The application opens http://localhost:3000 in the browser.
Change the URL on the browser to;
 ```
https://ah-robotics-frontend-sta-pr-21.herokuapp.com/

```

The browser will show like represented in screenshot _**image: 1**_
9. Click on the ```login``` button on the Navbar and enter a valid email **for demo purpose use**;

**Email:** ```steven.otieno1@gmail.com``` 
**Password:** ```test123#```

It will bring to a screen with **Profile** on the Navbar. See _**image: 2**_. 

10. Click on **Profile** this will bring you to _**image:3**_

You get to see: the count of
 **Followers** ```1```
 **Following** ```4```
<img width="356" alt="screenshot 2019-03-04 at 17 14 06" src="https://user-images.githubusercontent.com/39020475/53738703-f9507580-3ea0-11e9-9176-b54648da2b70.png">

11. To see the list of followers click on ```Followers``` button and then on ```Profile``` to revert.
12. To see the list of users you are following click on ```Following``` button and then on ```Profile``` to revert

13. Click on ```Home``` on the Navbar
Scroll the articles and click on a user with an interesting article that makes you want to follow them.
In this case as an example click on **By Mandela**. The action opens up their profile as in _**image:4**_
 
<img width="1126" alt="screenshot 2019-03-04 at 17 21 28" src="https://user-images.githubusercontent.com/39020475/53739216-4719ad80-3ea2-11e9-82a5-1dc2e1f50a9c.png">

14. To follow the user click on the **Follow** button.
15. To unfollow click on the **Unfollow** button and refresh page.

16. To run tests, open the terminal and run the following command:
```
npm test
``` 
17. To run tests and coverage, open the terminal and run the following command:
```
npm test -- --coverage
```
<img width="898" alt="screenshot 2019-03-05 at 08 51 56" src="https://user-images.githubusercontent.com/39020475/53783838-53464f00-3f24-11e9-8bc4-b224e4850d60.png">


#### Any background context you want to provide?
#### What are the relevant pivotal tracker stories?
[#162949047](https://www.pivotaltracker.com/story/show/162949047)
#### Screenshots (if appropriate)
_**Image: 1**_
<img width="1274" alt="screenshot 2019-03-04 at 16 47 24" src="https://user-images.githubusercontent.com/39020475/53737880-cad19b00-3e9e-11e9-923f-b74cf56126a8.png">

_**Image:2**_
<img width="1237" alt="screenshot 2019-03-04 at 17 02 25" src="https://user-images.githubusercontent.com/39020475/53738200-9b6f5e00-3e9f-11e9-8109-aee4879fda3d.png">

_**Image:3**_
<img width="1239" alt="screenshot 2019-03-04 at 17 08 59" src="https://user-images.githubusercontent.com/39020475/53738520-73ccc580-3ea0-11e9-93fc-fdbaaba9324b.png">

_**image:4**_
<img width="1243" alt="screenshot 2019-03-04 at 17 25 39" src="https://user-images.githubusercontent.com/39020475/53739430-c8714000-3ea2-11e9-824d-a402718eb3ee.png">





#### Questions:
